server/util/subdomain/subdomain.go

```diff
@@ -0,0 +1,42 @@
+package subdomain
+
+import (
+	"context"
+	"flag"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+)
+
+const subdomainKey = "subdomain"
+
+var (
+	enableSubdomainMatching = flag.Bool("app.enable_subdomain_matching", false, "If true, request subdomain will be taken into account when determining what request restrictions should be applied.")
+	defaultSubdomains       = flagutil.New("app.default_subdomains", []string{}, "List of subdomains that should not be handled as user-owned subdomains.")
```

Comment:
should this list match our other list of non-ownable subdomains?

Reply:
Which other list? This list would contain "app", "remote", "cache", "events", "cloud" for our setup.